From 29b85db81c3df444c069bf71a9f132cd8c3dd5a0 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 16 Dec 2023 13:01:47 -0500
Subject: [PATCH 006/105] init: Set generic package options

This change allows a generic way to set package options Example: HIDE_APPS="com.android.settings,com.aurora.services,com.termux,com.android.vending" RESTORE_APPS="com.android.settings,com.aurora.services,com.termux,com.android.vending"  Note: Be careful about what apps you disable, enable or hide. Some apps are required for other functions, like org.zeroxlab.util.tscal while others can not be disabled and only hidden, like com.android.settings  init: update generic package opts to take 1 or more packages  init: create arrays a different way for genpackopts  genpackopts: enable app then unhide  genpackopts: only hide and restore apps  Switching to using RESTORE_APPS  genopts: rever some of the last changes to allow pm command through  Only hide/unhide what we need  Do not add newlines to a single entry in the list
---
 init.sh | 52 ++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/init.sh b/init.sh
index 0d56c34..ea83077 100644
--- a/init.sh
+++ b/init.sh
@@ -990,6 +990,57 @@ function set_max_logd()
 	
 }
 
+function set_package_opts()
+{
+	# Set generic package options
+	# Example: HIDE_APPS="com.android.settings,com.aurora.services,com.termux,com.android.vending"
+	# 		   UNHIDE_APPS="com.android.settings,com.aurora.services,com.termux,com.android.vending"
+	#		   DISABLE_APPS="com.aurora.services,com.android.contacts,com.android.dialer"
+	# 		   ENABLE_APPS="com.aurora.services,com.android.contacts,com.android.dialer,com.android.messaging"
+	# 
+	# Note: Be careful about what apps you disable, enable or hide. Some apps are required for other
+	# functions, like org.zeroxlab.util.tscal while others can not be disabled and only hidden, 
+	# like com.android.settings
+	for c in `cat /proc/cmdline`; do
+        case $c in
+            *=*)
+                eval $c
+                if [ -z "$1" ]; then
+                    case $c in
+                        HIDE_APPS=*)
+                            hapackages="${HIDE_APPS#*=}"
+							hapackage_array=($(echo $hapackages | sed 's/,/ /g' | xargs))
+                            for hapackage in "${hapackage_array[@]}"; do
+								if [ ! -f /data/misc/bbconfig/$hapackage ]; then
+									echo "HIDE_APPS: $hapackage"
+									pm hide $hapackage
+									sleep 1
+									mkdir -p /data/misc/bbconfig
+									touch /data/misc/bbconfig/$hapackage
+								fi
+                            done
+                            ;;
+                        RESTORE_APPS=*)
+                            rapackages="${RESTORE_APPS#*=}"
+							rapackage_array=($(echo $rapackages | sed 's/,/ /g' | xargs))
+                            for rapackage in "${rapackage_array[@]}"; do
+								if [ -f /data/misc/bbconfig/$rapackage ]; then
+									echo "RESTORE_APPS: $rapackage"
+									pm unhide $rapackage
+									sleep 1
+									rm -rf /data/misc/bbconfig/$rapackage
+								fi
+                            done
+                            ;;
+                    esac
+                fi
+                ;;
+        esac
+    done
+	
+}
+
+
 
 function do_init()
 {
@@ -1104,6 +1155,7 @@ function do_bootcomplete()
 	fi
 
 	set_custom_package_perms
+	set_package_opts
 
 	post_bootcomplete
 }
-- 
2.34.1

